pkg/reconciler/reconcile.go

```diff
@@ -383,7 +381,7 @@ func (r *GenericServiceReconciler) syncAdapterService(ctx context.Context,
 }
 
 // findAdapter returns the adapter object for a given component instance if it exists.
-func findAdapter(genericReconciler interface{},
+func findAdapter[T kmeta.OwnerRefable](genericReconciler interface{},
```

Comment:
Please bear with me, I'm still trying to understand generics' use-cases. What's the practical value of having `findAdapter` implemented as a generic function instead of making it a method for different types of reconciler receivers?

Reply:
That's a great question! The single `findAdapter` function was my attempt (already before generics) to try reducing code duplication. But it resulted in something a bit cluttered.

To be clear, the `[T OwnerRefable]` on this line is only here because the function handles generic types, so the compiler forces you to have it here. I agree that it doesn't look right. I want to refactor that file a bit to use generics properly, but meanwhile I can split this function in this PR. I agree, it would make things less cluttered.

Reply:
Addressed in 7ea4ba93